test/test_autograd.py

```diff
@@ -4192,6 +4192,40 @@ def test_power_function(self):
         c.backward()
         self.assertEqual(b.grad, torch.tensor([-inf, 0., 0.]))
 
+    def test_nanprod_with_nans(self):
+        a = torch.randn(2, 2, 2, 2)
+        with torch.no_grad():
+            a[a < 0.2] = float('nan')
+        a.requires_grad = True
+
+        # No args
+        gradcheck(lambda x: x.nanprod(), a)
+        gradgradcheck(lambda x: x.nanprod(), a)
+
+        # Single dim
+        gradcheck(lambda x: x.nanprod(0), a)
+        gradgradcheck(lambda x: x.nanprod(0), a)
+
+        # With keep-dim
+        gradcheck(lambda x: x.nanprod(0, True), a)
+        gradgradcheck(lambda x: x.nanprod(0, True), a)
+
+    def test_nanprod_dtype(self):
+        inp = torch.randn(2, 2, 2, 2)
```

Comment:
nit: to reduce the test time, I think using `torch.randn(2, 2)` here and in the test above would be enough for all the cases.

Reply:
Done, thanks @albanD!